# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../nvi/hotfix.patch
# Copyright (C) 2019 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

$NetBSD: patch-.._ex_ex__script.c,v 1.1 2011/12/26 01:54:39 sbd Exp $

--- ../ex/ex_script.c.orig	2007-11-18 16:41:42.000000000 +0000
+++ ../ex/ex_script.c
@@ -23,8 +23,10 @@ static const char sccsid[] = "$Id: ex_sc
 #include <sys/select.h>
 #endif
 #include <sys/stat.h>
-#ifdef HAVE_SYS5_PTY
+#if 0 // def HAVE_SYS_STROPTS_H
 #include <sys/stropts.h>
+#elif 0 // defined(HAVE_STROPTS_H)
+#include <stropts.h>
 #endif
 #include <sys/time.h>
 #include <sys/wait.h>
@@ -750,6 +752,7 @@ ptys_open(int fdm, char *pts_name)
 		return (-5);
 	}
 
+#ifndef __linux__
 	if (ioctl(fds, I_PUSH, "ptem") < 0) {
 		close(fds);
 		close(fdm);
@@ -767,6 +770,7 @@ ptys_open(int fdm, char *pts_name)
 		close(fdm);
 		return (-8);
 	}
+#endif
 
 	return (fds);
 }

